meson.build

```diff
@@ -115,24 +115,35 @@ if not ['windows', 'cygwin'].contains(host_machine.system())
   endif
 endif
 
-soversion = host_machine.system() != 'windows' ? '1' : ''
+_files = [
+  files(
+    'lib/loadlibrary.c', 'lib/xmlparse.c', 'lib/xmlrole.c', 'lib/xmltok.c',
+    'lib/xmltok_impl.c', 'lib/xmltok_ns.c',
+  ),
+  config_h,
+]
 
-libexpat = library(
-  'expat',
-  [
-    files(
-      'lib/loadlibrary.c', 'lib/xmlparse.c', 'lib/xmlrole.c', 'lib/xmltok.c',
-      'lib/xmltok_impl.c', 'lib/xmltok_ns.c',
-    ),
-    config_h,
-  ],
-  vs_module_defs : 'lib/libexpat.def',
-  include_directories : include_directories('lib'),
-  dependencies : dep_libbsd,
-  version : '1.6.7',
-  soversion : soversion,
-  install : get_option('default_library') == 'shared',
-)
+if cc.get_id() == 'msvc'
+  libexpat = library(
+    'expat',
+    _files,
+    vs_module_defs : 'lib/libexpat.def',
+    include_directories : include_directories('lib'),
+    dependencies : dep_libbsd,
+    version : '1.6.7',
+    install : get_option('default_library') == 'shared',
+  )
+else
+  libexpat = library(
+    'expat',
+    _files,
+    include_directories : include_directories('lib'),
+    dependencies : dep_libbsd,
+    version : '1.6.7',
+    soversion : host_machine.system() != 'windows' ? '1' : '',
```

Comment:
Why do we need to special case `soversion`?

Reply:
on windows there is no soversion, it's just libexpat.dll, on *nix it is version. At least, that's what's in the windows install and what's on my linux machine.

Reply:
Does it produce an error? I think it would make sense to keep these 2 if branches as close as possible for any future reader.

Reply:
It doesn't that I know of (though someone may assume that they're going to get libexpat.dll not libexpat-1.dll, people do crazy things with cmake). I've always understood the goal of wraps to produce the same output that the original non-meson build system did, but maybe that's me misunderstanding?

Reply:
A okay, that's the difference. `soversion` on windows produces `libexpat-1.dll` and we want `libexpat.dll`. Do I understand it right?

Reply:
right. Their cmake build system creates a libexpat.dll, and if so_version is unset it will be libexpat-1.dll. On *nix though we want a libexpat.so.1.foo.bar